src/kubectlUtils.ts

```diff
@@ -0,0 +1,107 @@
+import * as vscode from "vscode";
+import { Kubectl } from "./kubectl";
+
+export type Cluster = {
+    name: string;
+    context: string;
+    user: string;
+    active: boolean;
+};
+
+export type Namespace = {
+    name: string;
+    active: boolean;
+};
+
+async function getKubeconfig(kubectl: Kubectl): Promise<any> {
+    const shellResult = await kubectl.invokeAsync("config view -o json");
+    if (shellResult.code !== 0) {
+        vscode.window.showErrorMessage(shellResult.stderr);
+        return null;
+    }
+    return JSON.parse(shellResult.stdout);
+}
+
+export async function getClusters(kubectl: Kubectl): Promise<Cluster[]> {
+    const kubectlConfig = await getKubeconfig(kubectl);
+    if (!kubectlConfig) {
+        return;
+    }
+    const currentContext = kubectlConfig["current-context"];
+    const contexts = kubectlConfig.contexts;
+    return contexts.map((c) => {
+        return {
+            name: c.context.cluster,
+            context: c.name,
+            user: c.context.user,
+            active: c.name === currentContext
+        };
+    });
+}
+
+export async function deleteCluster(kubectl: Kubectl, cluster: Cluster): Promise<boolean> {
+    const deleteClusterResult = await kubectl.invokeAsyncWithProgress(`config delete-cluster ${cluster.name}`, "Deleting cluster...");
+    if (deleteClusterResult.code !== 0) {
+        await vscode.window.showErrorMessage(`Failed to delete the specified cluster ${cluster.name} from the kubeconfig: ${deleteClusterResult.stderr}`);
+        return false;
+    }
+
+    const deleteUserResult = await kubectl.invokeAsyncWithProgress(`config unset users.${cluster.user}`, "Deleting user...");
+    if (deleteUserResult.code !== 0) {
+        await vscode.window.showErrorMessage(`Failed to delete user info for cluster ${cluster.name} from the kubeconfig: ${deleteUserResult.stderr}`);
+        return false;
+    }
+
+    const deleteContextResult = await kubectl.invokeAsyncWithProgress(`config delete-context ${cluster.context}`, "Deleting context...");
+    if (deleteContextResult.code !== 0) {
+        vscode.window.showErrorMessage(`Failed to delete the specified cluster's context ${cluster.context} from the kubeconfig: ${deleteContextResult.stderr}`);
+        return false;
+    }
+
+    vscode.window.showInformationMessage(`Successfully delete the associated cluster/context/user for cluster '${cluster.name}' from the kubeconfig.`);
+    return true;
+}
+
+export async function getNamespaces(kubectl: Kubectl): Promise<Namespace[]> {
+    const shellResult = await kubectl.invokeAsync("get namespaces -o json");
+    if (shellResult.code !== 0) {
+        vscode.window.showErrorMessage(shellResult.stderr);
+        return [];

```

Comment:
Ah, cool.  If returning an empty array on failure works for this function, can we also use that idiom in `getClusters` please?  Unless there is a specific reason callers need a separate failure signal in `getClusters`?